src/connection.js

```diff
@@ -262,6 +267,10 @@ class Connection extends EventEmitter {
       }
     }
 
+    if (this.config.domain && !this.config.userName && !this.config.password && SspiModuleSupported) {
+      this.config.options.useWindowsIntegratedAuth = true;
```

Comment:
@v-suhame The reasons here are sort of historical. username/password based NTLM support was added much before this PR and there are several customers out there who have come to rely on that. That implementation chose to use domain to distinguish between SQL authentication and NTLM authentication.

When I put together this PR to add more comprehensive support for Integrated Auth, I made sure to keep this compatible and not break any functionality. So I kept presence of domain as distinguishing factor between SQL authentication and Integrated Auth. Within that I chose the absence of username/password to switch to the new implementation.

Once this PR is merged, at some later point we can deprecate the username/password based NTLM as a breaking change and clean up the API a bit.

Hope this clarifies things.